plugins/inputs/net_response/net_response.go

```diff
@@ -191,8 +192,26 @@ func (n *NetResponse) Gather(acc telegraf.Accumulator) error {
 		n.Address = "localhost:" + port
 	}
 	if port == "" {
-		return errors.New("bad port")
+		return errors.New("bad port in config option address")
+	}
+
+	if err := choice.Check(n.Protocol, []string{"tcp", "udp"}); err != nil {
+		return fmt.Errorf("config option protocol: %w", err)
+	}
+
+	return nil
+}
+
+// Gather is called by telegraf when the plugin is executed on its interval.
+// It will call either UDPGather or TCPGather based on the configuration and
+// also fill an Accumulator that is supplied.
+func (n *NetResponse) Gather(acc telegraf.Accumulator) error {
+	// Prepare host and port
+	host, port, err := net.SplitHostPort(n.Address)
+	if err != nil {
+		return err
 	}
```

Comment:
I think this could also be done in `Init()`, but I don't want to hold up the PR just for this...

Reply:
It is done in Init to be able to error out. But should the `host` and `port` then be stored in the struct maybe?